man/ctags-lang-python.7.rst.in

```diff
@@ -0,0 +1,167 @@
+imported module is a reference tag; specify ``--extras=+r`` (or
+``--extras=+{reference}``) option for tagging it.  "roles:" field enabled with
+``--fields=+r`` is for recording the module is "imported" to the tag file.
+
+"input.py"
+.. code-block:: Python
+
+	import X1 as Y1
+
+"output.tags"
+with "--options=NONE -o - --extras=+r --fields=+rzK --fields-Python='+{nameref}' input.py"
+::
+
+	X1	input.py	/^import X1 as Y1$/;"	kind:module	roles:indirectlyImported
+	Y1	input.py	/^import X1 as Y1$/;"	kind:namespace	roles:def	nameref:module:X1
+
+"Y1" introduces a new name and is defined here. So "Y1" is tagged as a
+definition tag.  "X1" is imported in the way that its name cannot be used
+here. For letting know client tools that the name cannot be used,
+``indirectlyImported`` role is assigned for "X1".  "Y1" is the name for
+accessing objects defined in the module imported via "X1".  For recording this
+relationship, ``nameref:`` field is attached to the tag of "Y1".  Instead of
+``module`` kind, ``namespace`` kind is assigned to "Y1" because "Y1" itself
+doesn't define a module.
+
+"input.py"
+.. code-block:: Python
+
+	from X2 import *
+
+"output.tags"
+with "--options=NONE -o - --extras=+r --fields=+rzK input.py"
+::
+
+	X2	input.py	/^from X2 import \*$/;"	kind:module	roles:namespace
+
+The module is not defined here; it is defined in another file. So the tag for
+the imported module is a reference tag. Unlike "X0" in "import X0", "X2" may not
+be used because the names defined in "X2" can be used directly. The represent
+the difference ``namespace`` role is attached to "X2" instead of "imported".
+
+"input.py"
+.. code-block:: Python
+
+	from X3 import Y3
+
+"output.tags"
+with "--options=NONE -o - --extras=+r --fields=+rzKZ input.py"
+::
+
+	X3	input.py	/^from X3 import Y3$/;"	kind:module	roles:namespace
+	Y3	input.py	/^from X3 import Y3$/;"	kind:unknown	scope:module:X3	roles:imported
+
+"Y3" is a name for a language object defined in "X3" module. "scope:module:X3"
+attached to "Y3" represents this relation between "Y3" and "X3". ctags
+assigns ``unknown`` kind to "Y3" because ctags cannot know whether "Y3" is a
+class, a variable, or a function from the input file.
+
+"input.py"
+.. code-block:: Python
+
+	from X4 import Y4 as Z4
+
+"output.tags"
+with "--options=NONE -o - --extras=+r --fields=+rzKZ input.py"
+::
+
+	X4	input.py	/^from X4 import Y4 as Z4$/;"	kind:module	roles:namespace
+	Y4	input.py	/^from X4 import Y4 as Z4$/;"	kind:unknown	scope:module:X4	roles:indirectlyImported
+	Z4	input.py	/^from X4 import Y4 as Z4$/;"	kind:unknown	roles:def	nameref:unknown:Y4
+
+"Y4" is similar to "Y3" of "from X3 import Y3" but the name cannot be used here.
+``indirectlyImported`` role assigned to "Y4" representing this. "Z4" is the name for
+accessing the language object named in "Y4" in "X4" module. "nameref:unknown:Y4"
+attached to "Z4" and "scope:module:X4" attached to "Y4" represent the relations.
+
```

Comment:
How about put that little table in `parsers/python.c` here?

```

/* Roles related to `import'
 * ==========================
 * import X              X = (kind:module, role:imported)
 *
 * import X as Y         X = (kind:module, role:indirectlyImported),
 *                       Y = (kind:namespace, nameref:module:X)
 *                       ------------------------------------------------
 *                       Don't confuse the kind of Y with namespace role of module kind.
 *
 * from X import *       X = (kind:module,  role:namespace)
 *
 * from X import Y       X = (kind:module,  role:namespace),
 *                       Y = (kind:unknown, role:imported, scope:module:X)
 *
 * from X import Y as Z  X = (kind:module,  role:namespace),
 *                       Y = (kind:unknown, role:indirectlyImported, scope:module:X)
 *                       Z = (kind:unknown, nameref:unknown:Y) */
```

Reply:
I will do as you suggested. Thank you.

Reply:
I converted this to tables and put into the man  page.
The limitation of rst2man (or man itself), the comment is separated into multiple tables.